openflow13/nx_action.go

```diff
@@ -0,0 +1,953 @@
+package openflow13
+
+import (
+	"encoding/binary"
+	"errors"
+	"math"
+	"net"
+)
+
+const (
+	NxExperimenterID     = 0x00002320
+	NxActionHeaderLength = 10
+
+	NXAST_RESUBMIT         = 1
+	NXAST_SET_TUNNEL       = 2
+	NXAST_DROP_SPOOFED_ARP = 3
+	NXAST_SET_QUEUE        = 4
+	NXAST_POP_QUEUE        = 5
+	NXAST_REG_MOVE         = 6
+	NXAST_REG_LOAD         = 7
+	NXAST_NOTE             = 8
+	NXAST_SET_TUNNEL_V6    = 9
+	NXAST_MULTIPATH        = 10
+	NXAST_AUTOPATH         = 11
+	NXAST_BUNDLE           = 12
+	NXAST_BUNDLE_LOAD      = 13
+	NXAST_RESUBMIT_TABLE   = 14
+	NXAST_OUTPUT_REG       = 15
+	NXAST_LEARN            = 16
+	NXAST_EXIT             = 17
+	NXAST_DEC_TTL          = 18
+	NXAST_FIN_TIMEOUT      = 19
+	NXAST_CONTROLLER       = 20
+	NXAST_DEC_TTL_CNT_IDS  = 21
+	NXAST_PUSH_MPLS        = 23
+	NXAST_POP_MPLS         = 24
+	NXAST_SET_MPLS_TTL     = 25
+	NXAST_DEC_MPLS_TTL     = 26
+	NXAST_STACK_PUSH       = 27
+	NXAST_STACK_POP        = 28
+	NXAST_SAMPLE           = 29
+	NXAST_SET_MPLS_LABEL   = 30
+	NXAST_SET_MPLS_TC      = 31
+	NXAST_OUTPUT_REG2      = 32
+	NXAST_REG_LOAD2        = 33
+	NXAST_CNJUNCTION       = 34
+	NXAST_CT               = 35
+	NXAST_NAT              = 36
+	NXAST_CONTROLLER2      = 37
+	NXAST_SAMPLE2          = 38
+	NXAST_OUTPUT_TRUNC     = 39
+	NXAST_CT_CLEAR         = 43
+	NXAST_CT_RESUBMIT      = 44
+	NXAST_RAW_ENCAP        = 46
+	NXAST_RAW_DECAP        = 47
+	NXAST_DEC_NSH_TTL      = 48
+)
+
+type NXActionHeader struct {
+	*ActionHeader
+	Vender  uint32
+	Subtype uint16
+}
+
+func (a *NXActionHeader) Header() *ActionHeader {
+	return a.ActionHeader
+}
+
+func (a *NXActionHeader) NXHeader() *NXActionHeader {
+	return a
+}
+
+func (a *NXActionHeader) Len() (n uint16) {
+	return 10
```

Comment:
Please avoid magic numbers.

Reply:
I think this magic number is needed if message has a fixed length, as it could be used to malloc byte array in MarshalBinary and compare length of byte array in UnmarshalBinary. Besides, all existent code also use magic number for message length. I think this usage could keep the code style consistent.